src/coreclr/jit/lowerarmarch.cpp

```diff
@@ -1309,17 +1309,43 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
                 unsigned    simdSize        = node->GetSimdSize();
                 var_types   simdType        = Compiler::getSIMDTypeForSize(simdSize);
                 GenTree*    trueMask        = comp->gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
-                GenTree*    trueVal         = node;
-                GenTree*    falseVal        = comp->gtNewZeroConNode(simdType);
 
-                GenTreeHWIntrinsic* condSelNode =
-                    comp->gtNewSimdHWIntrinsicNode(simdType, trueMask, trueVal, falseVal, NI_Sve_ConditionalSelect,
-                                                   simdBaseJitType, simdSize);
+                switch (intrinsicId)
+                {
+                    // These are special cases where we need to handle the embedded mask differently
+                    // because ConditionalSelect expects a vector, but these APIs return a scalar.
+                    case NI_Sve_ExtractLastScalar:
+                    case NI_Sve_ExtractAfterLastScalar:
+                    {
+                        // Create the same node with an additional operand to pass the mask.
```

Comment:
@kunalspathak if you know an alternative way, I'm open to it.

Reply:
I think we should mark these with different flag, something like `EmbeddedMaskForScalarResult` and we can handle it here:

```c++
if (IsEmbeddedMaskedOperation())
{

}
else if (IsEmbeddedMaskForScalarResultOperation())
{
    
}
```

Reply:
Makes sense.